internal/saucecloud/xcuitest.go

```diff
@@ -23,6 +22,37 @@ type XcuitestRunner struct {
 	Project xcuitest.Project
 }
 
+// cache represents a store that can be used to cache return values of functions.
+type cache struct {
+	store map[string]string
+}
+
+func newCache() cache {
+	return cache{
+		store: make(map[string]string),
+	}
+}
+
+// lookup attempts to find the value for a key in the cache and returns if there's a hit, otherwise it executes and returns the closure fn
```

Comment:
Kinda sounds like it's returning `fn` 😅 , but it's returning its return values.

I can't get the wording quite right, but maybe this?
```suggestion
// lookup attempts to find the value for a key in the cache and returns if there's a hit, otherwise it executes the closure fn and returns its results.
```